### PR TITLE
(feature): add custom wallet-connect zengo module

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description
Based on the native wallet-connect module customized towards ZenGo wallet.
Only significant changes:

- Label
- Icon
- Open ZenGo app directly when on a mobile device

# Link issue
#219

# How Has This Been Tested?

Locally on desktop and mobile(android)

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/hooks/useActiveOnboard.ts [hunk 1]
```diff
@@ -186,7 +186,7 @@ export function useOnboardConnect():OnboardConnectProps {
 
     // disconnect
     if (!isConnected && previouslyConnected.length && (tried || activated)){
-      const isWalletConnect = previouslyConnected[0].label[0] === 'WalletConnect'
+      const isWalletConnect = previouslyConnected[0].label[0] === 'WalletConnect' || 'ZenGo'
```
Comment (sirpy) [inline src/hooks/useActiveOnboard.ts:189]: incorrect usage of ||

File: src/connectors/Zengo/index.ts [hunk 2]
```diff
@@ -0,0 +1,261 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
```
Comment (sirpy) [inline src/connectors/Zengo/index.ts:1]: what is this file based on?

Comment (L03TJ3) [reply src/connectors/Zengo/index.ts:1]: well based on... it's literally the wallet-connect implementation
https://github.com/blocknative/web3-onboard/blob/183ce59e08936849b026f11688429ffa77e93e2d/packages/walletconnect/src/index.ts#L1

this seemed the least time-intensive approach

File: src/hooks/useActiveOnboard.ts (hunk 1 above)
Comment (L03TJ3) [reply src/hooks/useActiveOnboard.ts]: right

**Review by sirpy — APPROVED** [review]

**Pull request merged**